server/gokbg3/grails-app/services/org/gokb/FTUpdateService.groovy

```diff
@@ -389,6 +398,76 @@ class FTUpdateService {
         }
 
         result.componentType = kbc.class.simpleName
+        result.tippTitleMedium = kbc.title ? kbc.title.medium : ""
+
+        result.accessStartDate = kbc.accessStartDate ? kbc.accessStartDate.toString() : ""
+        result.accessEndDate = kbc.accessEndDate ? kbc.accessEndDate.toString() : ""
+
+        result.subjectArea = kbc.subjectArea ?: ""
+        result.series = kbc.series ?: ""
+
+        if (kbc.title?.niceName == 'Book'){
+
+          // edition for eBooks
+          def edition = [:]
+          if (kbc.title?.editionNumber){
+            edition.number = kbc.title.editionNumber
+          }
+          if (kbc.title?.editionDifferentiator){
+            edition.differentiator = kbc.title.editionDifferentiator
+          }
+          if (kbc.title?.editionStatement){
+            edition.statement = kbc.title.editionStatement
+          }
+          if (!edition.isEmpty()){
+            result.titleEdition = edition
+          }
+
+          // simple eBook fields
+          result.titleVolumeNumber = kbc.title?.volumeNumber ?: ""
+          result.titleDateFirstInPrint = kbc.title?.dateFirstInPrint ?: ""
+          result.titleDateFirstOnline = kbc.title?.dateFirstOnline ?: ""
+          result.titleFirstEditor = kbc.title?.firstEditor ?: ""
+          result.titleFirstAuthor = kbc.title?.firstAuthor ?: ""
+          result.titleImprint = kbc.title?.imprint ?: ""
+        }
+
+        // title history for all title types
+        if (kbc.title?.hasProperty("historyEvents")){
+          result.titleHistory = []
+          kbc.title?.historyEvents?.each{ he ->
```

Comment:
`TitleInstance.historyEvents` is not defined. `title.titleHistory` returns a collection of directly connected history events for the title. `from` and `to` are arrays of TitleInstances, which would have to be mapped here as well. Events have an `id` but no `uuid`. The event `event.date` should be converted via the service as well.